src/coreclr/jit/gentree.cpp

```diff
@@ -19152,6 +19152,11 @@ bool GenTree::isContainableHWIntrinsic() const
             return true;
         }
 
+        case NI_AVX2_BroadcastScalarToVector256:
```

Comment:
Should we go ahead and cover the other broadcast APIs as well? That is, should we ensure this PR is "end to end" light-up of broadcast for 1 instruction (`addps`) and then we can simply add the flag as needed to the other intrinsics that support broadcast?

After `LowerHWIntrinsicCreate` we'll have one of:
* `GT_CNS_VEC`
* `NI_AVX512BW_BroadcastScalarToVector512`
* `NI_AVX512F_BroadcastScalarToVector512`
* `NI_AVX2_BroadcastScalarToVector256`
* `NI_AVX2_BroadcastScalarToVector128`
* `NI_SSE3_MoveAndDuplicate`

We can also have the following two, which are always from memory:
* `NI_AVX_BroadcastScalarToVector128`
* `NI_AVX_BroadcastScalarToVector256`

Reply:
What we expect from this PR is to light the overall framework of Embedded broadcast, and enable it for 1 instruction, `addps`. That implies, we will support vector add with vector length of 128, 256, 512, with float data type, (we may wanna include support for double, since in most sense, they share the same path.). So the follow-up work in this PR will be covering other broadcast APIs as pointed out in the comments.

Reply:
Won't the other broadcast APIs will be needed if we plan on supporting all 3 sizes (128, 256, and 512) of `float` in this PR?

Reply:
I believe those broadcast are needed, I will have a follow-up commit on this part in next few weeks.